cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -178,6 +178,14 @@ def create_parent_variable(parent_name)
         variables.build_option(name: 'PARENT_STACK', type: :string, value: parent_name)
       end
 
+      # Removes any `from: prompt` entries from variables
+      def disable_prompts
+        prompts = Set.new(%w(prompt service_link vault_cert_prompt))
+        variables.each do |option|
+          option.from.delete_if { |k,_| prompts.include?(k.to_s) }
```

Comment:
I think this counts as a hack.

I mean, something (?) like `option.from.each { |k, v| option.from[k] = DefaultResolver.new if prompts.include?(k.to_s) }` would work, but it's ugly...

Reply:
The defaults injection and value setting happen in the same manner. I don't see any better alternative except make some kind of global state to prompt resolver, which is not good either. Or implement this in opto like `group.new(disabled: [PrompResolver])`